pkg/controller/infinispan/templates/dependencies.gotmpl

```diff
@@ -0,0 +1,36 @@
+{{/* Dependencies download script for init container */}}
+set -e
+function retry {
+    local n=1
+    local max=5
+    local delay=1
+    while true; do
+        $@ && break || {
+            if [[ $n -lt $max ]]; then
+                ((n++))
+                echo "Download failed. Attempt $n/$max:"
+                sleep $delay
+            else
+                echo "Artifact download has failed after $n attempts."
+                exit 1
+            fi
+        }
+    done
+}
+function unpack {
+    if [[ ${2} == "" && ${1} =~ ".zip" || ${2} == "zip" ]]; then
+        unzip -oq ${1} && rm ${1}
+    fi
+    if [[ ${2} == "" && ${1} =~ ".tar.gz" || ${2} == "tgz" ]]; then
+        tar xf ${1} && rm ${1}
+    fi
+}
+cd {{ .MountPath }}
+{{- range $i, $artifact := .Artifacts }}
+    mkdir -p ./tmp && rm -rf ./tmp/* && cd ./tmp
+    retry "curl --insecure -LO {{ $artifact.Url }}"
+    FILENAME=$(ls -1 . | head -n1)
+    {{ hashCmd $artifact "$FILENAME" }}
+    cd .. && mv ./tmp/$FILENAME .
+    unpack {{ $artifact.Type }} $FILENAME
```

Comment:
I think we should do `rm -rf ./tmp` as the final step of the loop. That way the `./tmp` dir is deleted on the final loop (no currently the case.

So:

```
mkdir -p ./tmp
cd ./tmp
retry "curl --insecure -LO {{ $artifact.Url }}"
FILENAME=$(ls -1 . | head -n1)
{{ hashCmd $artifact "$FILENAME" }}
cd ..
mv ./tmp/$FILENAME .
unpack {{ $artifact.Type }} $FILENAME
rm -rf ./tmp
```

For readability I also think it's better if we don't chain the commands and have them as a invidivual lines now we're using a template.

Reply:
Yes, thanks, will fix.